### PR TITLE
Only create one instance of Features.None/PFxV1

### PR DESCRIPTION
**Pull request by lesaltzm**

### CODE REVIEW

**Review by CarlosFigueiraMSFT — DISMISSED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs [hunk 1]
```diff
@@ -96,9 +96,9 @@ public sealed class Features
         /// </summary>
         internal bool IsUserDefinedTypesEnabled { get; set; } = false;
 
-        internal static Features None => new Features();
+        internal static readonly Features None = new Features();
 
-        public static Features PowerFxV1 => new Features
+        public static readonly Features PowerFxV1 = new Features
```
Comment (CarlosFigueiraMSFT) [inline src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs:101]: The instances are readonly, but now it's possible that someone inadvertently change the meaning of those properties (i.e., by doing something like `Features.None.IsUserDefinedTypesEnabled = true`)

Comment (CarlosFigueiraMSFT) [reply src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs:101]: If we are to do this change, we should change this class to be immutable, and add a constructor where we can toggle those values.

Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs:101]: Yup, changing set to init on all these

**Review by MikeStall — APPROVED** [review]

**Comment by CarlosFigueiraMSFT** [comment]
>         internal Features(Features other)

I know this is not related to your changes, but since you're touching this file, can you update this copy constructor to add the missing properties?

AllowImpureNodeDelegation, AllowImpureNodeDelegation, AsTypeLegacyCheck, IsLegacyAnalysis, JsonFunctionAcceptsLazyTypes, IsLookUpReductionDelegationEnabled

---
Refers to: src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs:121 in 909a83b. [](commit_id = 909a83b6b2c5837e9258c16d9f09861f51ce1973, deletion_comment = False)

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Features.get_PowerFxV1

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs [hunk 2]
```diff
@@ -707,7 +707,7 @@ public CheckContextSummary ApplyGetContextSummary()
             var summary = new CheckContextSummary
             {
                 AllowsSideEffects = allowSideEffects,
-                IsPreV1Semantics = isV1,
+                IsPreV1Semantics = !isV1,
```
Comment (Shpakh-MSFT) [inline src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs:710]: >!isV1 [](http://example.com/codeflow?start=35&length=5)

Was this a bug?

**Review by Shpakh-MSFT — APPROVED** [review]
:shipit:

File: src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs (hunk 2 above)
Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs:710]: Yes, discussed with Mike and Hoa. The equality check above was always false before

**Comment by LucGenetier** [comment]
Public API changes: <br />❌CP0002 M:Microsoft.PowerFx.Features.get_PowerFxV1

**Comment by MikeStall** [comment]
Somehow, this change is causing a regression and this is now failing:

            PowerFxConfig config = new PowerFxConfig(); // Pass in per engine
            Assert.Equal(config.Features.SupportColumnNamesAsIdentifiers, true);

It changes a static property to a static field - probably some .Net race condition around static field initialization.

**Pull request merged**